config/software/ruby.rb

```diff
@@ -104,10 +104,7 @@
   # wrlinux7/ios_xr build boxes from Cisco include libssp and there is no way to
   # disable ruby from linking against it, but Cisco switches will not have the
   # library.  Disabling it as we do for Solaris.
-  #
-  # This will be changed to use a chef-sugar helper method once
-  # sethvargo/chef-sugar#116 is available in a release.
-  if ohai['platform'] == 'ios_xr' && version.to_f >= 2.1
+  if ios_xr? && version.to_f >= 2.1
```

Comment:
@scotthain is indeed correct, but we can still use a constraint check and pass in the version.  I'll update it.